compiler/rustc_const_eval/src/interpret/cast.rs

```diff
@@ -366,22 +366,6 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             }
             (&ty::Adt(def_a, _), &ty::Adt(def_b, _)) => {
                 assert_eq!(def_a, def_b);
-                if def_a.is_box() || def_b.is_box() {
-                    if !def_a.is_box() || !def_b.is_box() {
-                        span_bug!(
-                            self.cur_span(),
-                            "invalid unsizing between {:?} -> {:?}",
-                            src.layout.ty,
-                            cast_ty.ty
-                        );
-                    }
-                    return self.unsize_into_ptr(
```

Comment:
Uh... are you saying this will just work if we recurse into the type?
I guess we'll go below the Unique, NonNull, and then it's just a normal raw pointer...

Very nice!

Reply:
Yeah, it just works. I'm not sure why Box was ever special-cased. I guess this code must be from before CoerceUnsized existed?

Reply:
This is originally copied from codegen, I think. Doesn't codegen still do something similar?

Reply:
I don't see any box special cases in codegen for unsizing casts, so not any more I guess.

Reply:
Yeah they probably removed it, making the same observation you now did.